src/v/cloud_topics/read_path/placeholder_extent.h

```diff
@@ -39,29 +34,24 @@ struct hydrated_L0_object {
 // The extent can be hydrated (the data is moved from the cloud
 // storage to disk) or materialized (data is moved to the main
 // memory).
-struct placeholder_extent {
-    model::offset base_offset;
-    dl_placeholder placeholder;
-    ss::lw_shared_ptr<hydrated_L0_object> L0_object;
+struct materialized_extent {
+    extent_meta meta;
+    iobuf object;
```

Comment:
commit message says we are using extent_meta in placeholder_extent, but it seems like we are also either renaming or replacing it with something new? please explain what is going on in the commit message.

Reply:
fixed that